share/wake/lib/core/syntax.wake

```diff
@@ -27,6 +27,16 @@ export def (argument: a).(memberFn: a => b): b =
 export def (argument: a) | (pipeFn: a => b): b =
     pipeFn argument
 
+# Like '|' above but rmapPass the pipeFn to enter a "Result pipeline"
+# rmapPass (catWith " " _) (Pass (seq 10)) = Pass (seq 10) |> catWith " "
+export def (argument: Result a b) |> (pipeFn: a => Result c b): Result c b =
+    rmapPass (\a pipeFn a) argument
+
+# Like '|' above but rmapPass the pipeFn to enter a "Result pipeline"
+# rmapPass (catWith " " _) (Pass (seq 10)) = Pass (seq 10) |> catWith " "
+export def (argument: Result a b) |< (pipeFn: a => c): Result c b =
```

Comment:
@ag-eitilt I'm not 100% sure on including `|<`. `|>` should probably only be used where is was previously designed for but assuming that isn't followed writing stuff without `|<` is quite painful when the pipe function doesn't return a result.

Example:
`(seq 10) | Pass |> (\xs (map str xs) | Pass) |> (\x catWith " " x | Pass)` is messy and painful. You can't use stuff like `map str _` in this context. Adding `|<` resolves the issue with the tradeoff of extra cognitive load.
`(seq 10) | Pass |< (map str _) |< catWith " "`

Reply:
I think it'll also be helpful for us if we can name/describe `|>` and `|<`. Right now I was thinking "continue the result" for `|>` and maybe something like "insert/inject a result" for `|<`